### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by ctriant**

### CODE REVIEW

**Comment by ctriant** [comment]
@rohe could you please have a look at my [comment](https://github.com/IdentityPython/idpy-oidc/pull/38#discussion_r1034512918) in #38 when possible?

File: src/idpyoidc/server/oauth2/introspection.py [hunk 1]
```diff
@@ -103,7 +104,7 @@ def process_request(self, request=None, release: Optional[list] = None, **kwargs
             _session_info = _context.session_manager.get_session_info_by_token(
                 request_token, grant=True
             )
-        except (UnknownToken, WrongTokenClass):
+        except (UnknownToken, WrongTokenClass, ToOld):
```
Comment (c00kiemon5ter) [inline src/idpyoidc/server/oauth2/introspection.py:107]: This change is important to properly mark token with `active: false` on the introspection endpoint, when the token has expired.

Comment (c00kiemon5ter) [reply src/idpyoidc/server/oauth2/introspection.py:107]: (Maybe this should be split into its own PR, along with the relevant test case.)

File: src/idpyoidc/server/client_authn.py [hunk 2]
```diff
@@ -262,14 +262,15 @@ def _verify(
         request: Optional[Union[dict, Message]] = None,
         authorization_token: Optional[str] = None,
         endpoint=None,  # Optional[Endpoint]
+        get_client_id_from_token=None,
```
Comment (peppelinux) [inline src/idpyoidc/server/client_authn.py:265]: is this a pointer to a function/method?

wouldn't it be better Optional[...]?

**Comment by ctriant** [comment]
I will try to break this MR into smaller ones, to facilitate the review process.
Mainly the changes regard to:
- Client authentication
- Token exchange enhancements
- Introspection

**Comment by ctriant** [comment]
Closing this MR in favor of #47, #48 and #49.
Trying to elaborate on each one separately. I'll transfer all comments to the appropriate MR

**Pull request closed without merging**